nntrainer/graph/network_graph.cpp

```diff
@@ -496,26 +504,28 @@ LayerNode &NetworkGraph::getLayerNode(const std::string &layer_name) {
   throw std::invalid_argument("Cannot find Layer");
 }
 
-int NetworkGraph::setEdge() {
-  int status = ML_ERROR_NONE;
+void NetworkGraph::setEdge(unsigned int adj_idx) {
+  std::list<LayerNode>::iterator iter = adj[adj_idx].begin();
```

Comment:
check if adj_idx exist in `adj`?

Reply:
setEdge is a private function called by internal functions only. 
The way it is called ensures that `adj_idx < adj.size()`.